app/components/PathToDocument.js

```diff
@@ -6,36 +6,30 @@ import styled from "styled-components";
 import type { DocumentPath } from "stores/CollectionsStore";
 import Collection from "models/Collection";
 import Document from "models/Document";
+import Checkbox from "components/Checkbox";
 import CollectionIcon from "components/CollectionIcon";
 import Flex from "components/Flex";
 
 type Props = {
   result: DocumentPath,
   document?: ?Document,
   collection: ?Collection,
-  onSuccess?: () => void,
+  checked?: boolean,
+  onSelect?: (DocumentPath) => void,
   style?: Object,
   ref?: (?React.ElementRef<"div">) => void,
 };
 
 @observer
 class PathToDocument extends React.Component<Props> {
-  handleClick = async (ev: SyntheticEvent<>) => {
+  handleClick = (ev: SyntheticEvent<>) => {
```

Comment:
I don't think this method is needed at all, could just attach `this.props.onSelect` directly – afterall we're passing in the `result` prop to just pass it back out again.

This component is super old, feel free to tear it apart of rebuild as you see fit.